File: examples/ruby/guides/routing/mock_classifier.rb
# frozen_string_literal: true

def mock_classify(message)
  lower = message.downcase
  return 'support' if %w[bug error help].any? { |w| lower.include?(w) }
  return 'sales' if %w[price buy plan].any? { |w| lower.include?(w) }

  'other'
end

def mock_reply(message, role)
  case role
  when 'support'
    "[Support] I can help with that technical issue. Let me look into: #{message}"
  when 'sales'
    "[Sales] Great question about pricing! Here's what I can tell you about: #{message}"
  else
    "[General] Thanks for reaching out. Regarding: #{message}"
  end
end

File: examples/typescript/guides/routing/mock-classifier.ts
export function mockClassify(message: string): string {
  const lower = message.toLowerCase();
  if (lower.includes('bug') || lower.includes('error') || lower.includes('help')) return 'support';
  if (lower.includes('price') || lower.includes('buy') || lower.includes('plan')) return 'sales';
  return 'other';
}

export function mockReply(message: string, role: string): string {
  switch (role) {
    case 'support':
      return `[Support] I can help with that technical issue. Let me look into: ${message}`;
    case 'sales':
      return `[Sales] Great question about pricing! Here's what I can tell you about: ${message}`;
    default:
      return `[General] Thanks for reaching out. Regarding: ${message}`;
  }
}
